forum/qa-plugin/block-pw/block-pw-layer.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+class qa_html_theme_layer extends qa_html_theme_base
+{
+    private $isLogged;
+    
+    public function nav_list($navigation, $class, $level=null) // cannot use `: void` type-hint
+    {
+        $user = qa_request_parts()[1] ?? '';
+        $dbUser = qa_db_select_with_pending(qa_db_user_account_selectspec($user, false));
+
+        if (qa_clicked('douserblock') || qa_clicked('douserunblock')) {
+            $this->performFormAction(qa_get_logged_in_userid(), $dbUser['userid']);
+        }
+
+        $this->prepareNavigation($class, $dbUser['handle'], $navigation);
+        $this->prepareProfileButtons($class, qa_get_logged_in_userid(), $dbUser);
+        $this->changePrivateMessageButton(qa_get_logged_in_userid(), $dbUser);
+        
+        parent::nav_list($navigation, $class, $level);
+    }
+    
+    private function performFormAction($loggedInId, $profileUserId): void
+    {
+        if (qa_clicked('douserblock')) {
+            qa_db_query_sub('INSERT INTO `^blockedpw` VALUES (#, #)', $loggedInId, $profileUserId);
```

Comment:
`qa_db_query_sub()` zwraca dane lub error, który można obsłużyć - warto to obsłużyć w jakiś sposób. W końcu jeśli wystąpi błąd z bazą -> to tu się wywali.